### ISSUE TITLE
Add an async example with Algolia

### ISSUE THREAD

**Issue by Haroenv** [effektif/react-mentions#190]
I'd like to make an example of how to use this with [React InstantSearch](https://github.com/algolia/react-instantsearch), and was wondering if an example similar to [ant-mentions](https://github.com/algolia/react-instantsearch/tree/master/packages/react-instantsearch/examples/autocomplete) could be useful here 😄

**Comment by jfschwarz** [comment]
Absolutely! This would make a great example and we would be more than happy to accept a PR for this. Let me know if you need any help.

**Comment by frontendphil** [comment]
@Haroenv did you make any progress or can we close this issue?

**Comment by Haroenv** [comment]
Sorry, this issue slipped my mind, I can still make the example similar to the one linked above.
